### DISCUSSION TITLE
Include skipped tests in the output

### DISCUSSION THREAD

**Discussion by bsipocz** [pytest-dev/pytest#5551] [General] [answered]
Currently if a whole test file is skipped at discovery time with either `pytest.skip(allow_module_level=True)` or with `pytest.importorskip()` it's not listed in the report. Having it listed there as skipped would be still useful information.

**Comment by nicoddemus** [comment, marked as answer]
Hi @bsipocz,

Thanks for following up with an issue.

I don't think this is simple to achieve today, I'm afraid. Both `pytest.skip` and `pytest.importorskip` work by raising an *exception* during import, which prevents the rest of the module from being processed. 

Note that this behavior might still be achieved today through this slightly more verbose code:

```python
try:
    import asd.util
except ImportError as e:
    pytestmark = pytest.mark.skip(reason=str(e))
```

Instead of:

```python
asd = pytest.importorskip('asd.util')
```

**Comment by bilderbuchi** [comment]
Just came here because I also hit this behaviour - it was surprising to me that a _skipped_ module seems to not be _collected_, as its tests don't show up in the report - it's as if the whole test module is not there. This does not mesh with my understanding of skipping in pytest. I was checking if I made a mistake in the test harness structure or whatnot which made the tests undiscoverable, until I started googling.

(I used `pytest.skip(allow_module_level=True)`).

**Comment by nicoddemus** [comment]
@bilderbuchi but does the workaround of using `pytestmark` work for you?

**Reply by bilderbuchi** [reply]
Sorry, it seems I never got the notification/ping (maybe because this is a "discussion" now?)- yes, thank you, the workaround works. :-)

**Reply by nicoddemus** [reply]
No worries; weird I get email notifications from discussions normally as I get with issues.
